### PR TITLE
global config validation

### PR DESCRIPTION
**Pull request by ChomickiM**
Updating the front end code for Global Configuration to use our new validation format

### CODE REVIEW

**Review by psantos1113 — APPROVED** [review]

File: src/main/js/store/actions/globalConfiguration.js [hunk 1]
```diff
@@ -153,18 +157,39 @@ function clearFieldErrors() {
...
+    errorHandlers.push(HTTPErrorUtils.createBadRequestHandler(defaultHandler));
+    errorHandlers.push(HTTPErrorUtils.createPreconditionFailedHandler(defaultHandler));
+    errorHandlers.push(HTTPErrorUtils.createDefaultHandler(defaultHandler));
+
+    return HTTPErrorUtils.createHttpErrorHandler(errorHandlers);
 }
 
-export function refreshConfig(id) {
+export function
+refreshConfig(id) {
```
Comment (bamandel) [inline src/main/js/store/actions/globalConfiguration.js:192]: Do we prefer a multiline method signature?

**Review by bamandel — APPROVED** [review]

File: src/main/js/store/actions/globalConfiguration.js [hunk 2]
```diff
@@ -258,17 +304,29 @@ export function updateConfig(config) {
...
+                    response.json()
+                        .then((responseData) => {
+                            const newId = responseData.id;
+                            const updatedConfig = FieldModelUtilities.updateFieldModelSingleValue(config, 'id', newId);
+                            dispatch(configUpdated(updatedConfig));
+                            dispatch(refreshConfig(newId));
+                        });
+                } else {
+                    dispatch(refreshConfig(id));
+                }
```
Comment (gkillough) [inline src/main/js/store/actions/globalConfiguration.js:318]: You could remove the else (for the response.status check) and only need to call refresh once: 
```suggestion
                        });
                }
                dispatch(refreshConfig(id));
```

**Review by gkillough — APPROVED** [review]

File: src/main/js/store/actions/globalConfiguration.js (hunk 2 above)
Comment (gkillough) [reply src/main/js/store/actions/globalConfiguration.js:318]: Ah, but you'll have to create an id variable outside of the if-statement and set it for a 201.

File: src/main/js/store/actions/globalConfiguration.js (hunk 1 above)
Comment (jamesrichard91) [reply src/main/js/store/actions/globalConfiguration.js:192]: Why is `export function` and `refreshConfig(id) {` on different lines?

Comment (ChomickiM) [reply src/main/js/store/actions/globalConfiguration.js:192]: This was a mistake, I can update it

**Pull request merged**